### PR TITLE
Make OffsetDependentTypesHelper use standardizeString

### PR DESCRIPTION
**Pull request by mernst**
~Merge after #4195.  (Can review now, examining diffs without those in #4195.)~

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/common/value/ValueCheckerUtils.java [hunk 1]
```diff
@@ -389,4 +401,129 @@ public static int getMinLen(Tree tree, ValueAnnotatedTypeFactory valueAnnotatedT
         AnnotatedTypeMirror minLenType = valueAnnotatedTypeFactory.getAnnotatedType(tree);
         return valueAnnotatedTypeFactory.getMinLenValue(minLenType);
     }
+
+    /**
+     * Optimize the given JavaExpression, using Value Checker annotations.
```
Comment (kelloggm) [inline framework/src/main/java/org/checkerframework/common/value/ValueCheckerUtils.java:406]: `Optimize` is a vague name. I think I'm fine with it here as the name of this method - it conveys the right intuition for what this method does - but this Javadoc needs to define it. I think the specific meaning here is "constant fold any parts of the given expression that the factory has exact values for", based on reading the code.

**Review by kelloggm — CHANGES_REQUESTED** [review]
I only reviewed the parts that aren't in #4195

File: framework/src/main/java/org/checkerframework/common/value/ValueCheckerUtils.java [hunk 2]
```diff
@@ -389,4 +401,131 @@ public static int getMinLen(Tree tree, ValueAnnotatedTypeFactory valueAnnotatedT
...
+            JavaExpression optLeft = optimize(e.getLeft(), factory);
+            JavaExpression optRight = optimize(e.getRight(), factory);
+            if (e.getLeft() == optLeft && e.getRight() == optRight) {
+                return e;
+            } else {
+                return new BinaryOperation(e.getType(), e.getOperationKind(), optLeft, optRight);
+            }
+
+        } else if (je instanceof ClassName) {
+            ClassName e = (ClassName) je;
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/common/value/ValueCheckerUtils.java:449]: This cast and assignment aren't needed.

**Review by smillst — APPROVED** [review]
Does this fix https://github.com/typetools/checker-framework/issues/3475?

**Comment by mernst** [comment]
> Does this fix #3475?

Not quite.  There is still one difference in `OffsetDependentTypesHelper`.  I think that code can be moved into the superclass in a future pull request.

**Pull request merged**